src/wh_crypto.c

```diff
@@ -44,6 +45,49 @@
 
 #include "wolfhsm/wh_crypto.h"
 
+#ifndef NO_RSA
+int wh_Crypto_SerializeRsaKey(RsaKey* key, uint16_t max_size,
+        uint8_t* buffer, uint16_t *out_size)
+{
+    int ret = 0;
+    int der_size = 0;
+
+    if (    (key == NULL) ||
+            (buffer == NULL)) {
+        return WH_ERROR_BADARGS;
+    }
+
+    der_size = wc_RsaKeyToDer(key, (byte*)buffer, (word32)max_size);
+    if (der_size >= 0) {
+        ret = 0;
+        if (out_size != NULL) {
+            *out_size = der_size;
+        }
+    } else {
+        /* Error serializing.  Clear the buffer */
+        ret = der_size;
+        memset(buffer, 0, max_size);
+    }
+    return ret;
+}
+
+int wh_Crypto_DeserializeRsaKey(uint16_t size, const uint8_t* buffer,
+        RsaKey* key)
+{
+    int ret;
+    word32 idx = 0;
+
+    if (    (size == 0) ||
+            (buffer == NULL) ||
+            (key == NULL)) {
+        return WH_ERROR_BADARGS;
+    }
+    /* Deserialize the RSA key */
+    ret = wc_RsaPrivateKeyDecode(buffer, &idx, key, size);
```

Comment:
this only handles private keys - a trickier problem to determine if the DER is public or private. Not sure we can do that in one pass if no metadata is present indicating whether public or private. Perhaps we try public first, which I think checks the ASN.1 OID, and if that fails try private?